### PR TITLE
Use Connect instead of gRPC

### PR DESCRIPTION
**Pull request by dylan-bourque**
This PR swaps gRPC for Connect in both the server and CLI implementations.  For compatibility with running instances of the Perseus service, we enable the gRPC wire protocol for the client.  We have also replaced `grpc-gateway` with Vanguard (connectrpc.com/vanguard) for JSON/REST support in the server's API.

I have done compatibility testing using an old (gRPC) client hitting a new (Connect) server, a new client hitting an old server, and a new client hitting a new server.

The generated metrics are slightly different due to differences between the gRPC and Connect runtimes, but the things being monitored are the same.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,8 @@ BUILD_TIME_TOOLS =\
 	github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway\
 	github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2 \
     google.golang.org/protobuf/cmd/protoc-gen-go \
-    google.golang.org/grpc/cmd/protoc-gen-go-grpc
+    google.golang.org/grpc/cmd/protoc-gen-go-grpc \
```
Comment (bufdev) [inline Makefile:29]: Nit: inconsistent indentation

File: buf.gen.yaml [hunk 2]
```diff
@@ -10,16 +10,12 @@ plugins:
...
-    opt:
-      - paths=source_relative
-      - allow_repeated_fields_in_body=true
-    out: ./perseusapi
   - name: openapiv2
     opt:
       - allow_repeated_fields_in_body=true
-    out: ./docs
\ No newline at end of file
+    out: ./docs
```
[the remark this answers is not in the document]
Comment (dylan-bourque) [reply buf.gen.yaml]: Done

File: client_config.go [hunk 3]
```diff
@@ -84,40 +79,18 @@ func readClientConfigFlags(fset *pflag.FlagSet) []clientOption {
...
-		return nil, err
+		opts = append(opts, httplb.WithTLSConfig(&tlsc, 0))
 	}
 
-	// create and return the client
-	return perseusapi.NewPerseusServiceClient(conn), nil
+	httpc := httplb.NewClient(opts...)
+	// we include WithGRPC() so that the CLI can hit an existing gRPC-based server instance
+	// - this may be removed at some point in the future
+	cc := perseusapiconnect.NewPerseusServiceClient(httpc, conf.serverAddr, connect.WithGRPC())
```
[the remark this answers is not in the document]
Comment (dylan-bourque) [reply client_config.go]: Done

**Comment by dylan-bourque** [comment]
Ran `git pull --rebase origin main` to sync with unrelated Dependabot updates and force-pushed

File: buf.gen.yaml [hunk 4]
```diff
@@ -9,17 +9,13 @@ managed:
...
     opt:
       - paths=source_relative
-    out: ./perseusapi
-  - name: grpc-gateway
-    opt:
-      - paths=source_relative
-      - allow_repeated_fields_in_body=true
-    out: ./perseusapi
+    out: perseusapi
   - name: openapiv2
```
Comment (dylan-bourque) [inline buf.gen.yaml:17]: A note here: Vanguard does not currently have its own OpenAPI/Swagger generator so we're still using gRPC Gateway's `protoc-gen-openapiv2`.

Comment (dylan-bourque) [reply buf.gen.yaml:17]: Thanks for the tip.  I'll take a look and see if that's an acceptable replacement.

File: internal/server/connect.go
Comment (dylan-bourque) [inline internal/server/connect.go:1]: This file replaces `grpc.go` with the new, Connect-based implementation of the same service API.

**Review by bndw — APPROVED** [review]
LGTM. Please update the README section on [service architecture](https://github.com/CrowdStrike/perseus?tab=readme-ov-file#service-architecture) to reflect the change from grpc-gateway -> connect.

File: buf.gen.yaml (hunk 4 above)
Comment (dylan-bourque) [reply buf.gen.yaml:17]: Give it a whirl and `protoc-gen-connect-openapi` outputs "pure" OpenAPI, which is not quite the same thing as Swagger.  Haven't had time to do a deep comparison to see how different things are and whether or not those differences matter for things like Swagger UI.

Gonna leave that for a follow-up PR, though.

**Review by alee792 — APPROVED** [review]
Wrote out a few comments but realized I was more reviewing and asking questions about the connect API and not your actual code lol. Looks great!

**Pull request merged**